pre/file/string.hpp

```diff
@@ -91,4 +91,21 @@ namespace pre::file {
     }
   }
 
+  inline void from_string(const std::string &path, const std::string &content, std::error_condition &ec,const boost::filesystem::perms& perms,  boost::system::error_code& ec_perms) {
+    from_string(path, content, ec);
+    boost::filesystem::permissions(path, perms, ec_perms);
+  }
```

Comment:
I don't think it's a great idea to have 2 error code types, that there are 2 parameters : why not. But it would be better there would be only one holding both codes and the function could just return early on error.